Jenkinsfile

```diff
@@ -1,10 +1,17 @@
+def parseSemanticVersion(String version) {
+  def semanticVersionRegex = /^(((([0-9]+)\.[0-9]+)\.[0-9]+)(?:\-.+)?)$/
+  def (_, prerelease, patch, minor, major) = (version =~ semanticVersionRegex)[0]
+  return [major, minor, patch, prerelease]
+}
```

Comment:
Tested this in a repl.

Try executing the following in https://www.jdoodle.com/execute-groovy-online/:
```groovy
def parseSemanticVersion(String version) {
  def semanticVersionRegex = /^(((([0-9]+)\.[0-9]+)\.[0-9]+)(?:\-.+)?)$/
  def (_, prerelease, patch, minor, major) = (version =~ semanticVersionRegex)[0]
  return [major, minor, patch, prerelease]
}

def printVersion(String version) {
  def (major, minor, patch, prerelease) = parseSemanticVersion(version)
  println "== ${version} =="
  println "Major: ${major}"
  println "Minor: ${minor}"
  println "Patch: ${patch}"
  println "Prerelease: ${prerelease}"
}

printVersion("123.456.789-pre.1")
print "\n"
printVersion("123.456.789")
```